src/System.Private.CoreLib/shared/System/IO/Stream.cs

```diff
@@ -1270,6 +1237,9 @@ public override int ReadByte()
 
             public override IAsyncResult BeginRead(byte[] buffer, int offset, int count, AsyncCallback callback, Object state)
             {
+#if CORERT
+                throw new PlatformNotSupportedException(); // TODO: https://github.com/dotnet/corert/issues/3251
```

Comment:
In corert we have `NotImplementedException` (refer to [this](https://github.com/dotnet/corert/blob/ed0b32ee6b2c3c9d7c9e1c21ca019cd60278f861/src/System.Private.CoreLib/src/System/IO/Stream.cs#L723)). Do we want to switch behavior to `PlatformNotImplementedException`?